### PR TITLE
Possibility to read isCesnetEligible attribute from LDAP

### PR DESCRIPTION
**Pull request by BaranekD**

### CODE REVIEW

File: lib/Auth/Process/IsCesnetEligible.php [hunk 1]
```diff
@@ -45,24 +52,39 @@ class IsCesnetEligible extends \SimpleSAML\Auth\ProcessingFilter
      */
     private $rpcConnector;
 
+    /**
+     * @var LdapConnector\
```
Comment (vyskocilpavel) [inline lib/Auth/Process/IsCesnetEligible.php]: Remove the backslash

File: lib/Auth/Process/IsCesnetEligible.php [hunk 2]
```diff
@@ -45,24 +52,39 @@ class IsCesnetEligible extends \SimpleSAML\Auth\ProcessingFilter
...
-        $this->cesnetEligibleLastSeenAttr = $config[self::CESNET_ELIGIBLE_LAST_SEEN_ATTR];
+        $this->rpcAttrName = $config[self::RPC_ATTRIBUTE_NAME];
+        $this->interface = $config[self::INTERFACE_PROPNAME];
+
+        if ($this->interface === self::LDAP) {
+            $this->ldapAttrName = $config[self::LDAP_ATTRIBUTE_NAME];
+            if (!empty($this->ldapAttrName)) {
+                $this->ldapConnector = new AdapterLdap();
+            } else {
+                $this->interface = self::RPC;
```
Comment (vyskocilpavel) [inline lib/Auth/Process/IsCesnetEligible.php]: Please add warning that ldapAttrName is missing and RPC interface will be used

File: lib/Auth/Process/IsCesnetEligible.php [hunk 3]
```diff
@@ -104,32 +126,53 @@ public function process(&$request)
...
         } catch (Exception $ex) {
             Logger::warning("cesnet:IsCesnetEligible - " . $ex->getMessage());
         }
 
-        if ($this->cesnetEligibleLastSeen['value'] != null) {
-            $request['Attributes'][$this->attrName] = array($this->cesnetEligibleLastSeen['value']);
+        if ($this->cesnetEligibleLastSeenValue != null) {
+            $request['Attributes'][$this->returnAttrName] = array($this->cesnetEligibleLastSeenValue);
+            Logger::debug(
+                "cesnet:IsCesnetEligible - Attribute " . $this->returnAttrName . " was sent with value " .
```
Comment (vyskocilpavel) [inline lib/Auth/Process/IsCesnetEligible.php]: There will be better use msg that attribute was setted to value.

**Review by vyskocilpavel — CHANGES_REQUESTED** [review]

File: lib/Auth/Process/IsCesnetEligible.php (hunk 1 above)
Comment (BaranekD) [reply lib/Auth/Process/IsCesnetEligible.php]: Done

File: lib/Auth/Process/IsCesnetEligible.php (hunk 2 above)
Comment (BaranekD) [reply lib/Auth/Process/IsCesnetEligible.php]: Done

File: lib/Auth/Process/IsCesnetEligible.php (hunk 3 above)
Comment (BaranekD) [reply lib/Auth/Process/IsCesnetEligible.php]: Done

**Review by dBucik — APPROVED** [review]

**Review by vyskocilpavel — APPROVED** [review]

**Pull request merged**